CHANGELOG.md

```diff
@@ -1,7 +1,22 @@
 x.y.z Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
-* None.
+* Add `createdAt`, `updatedAt`, `expiresAt` and `timeToLive` properties to
+  `RLMSyncSubscription`/`SyncSubscription`. This properties will be `nil` for
+  subscriptions created with older versions of Realm, but will be automatically
+  populated for newly-created subscriptions.
+* Add support for transient subscriptions by setting the `timeToLive` when
+  creating the subscription. The next time a subscription is created or updated
+  after that time has elapsed the subscription will be automatically removed.
+* Add support for updating existing subscriptions with a new query or limit.
+  This is done by passing `update: true` (in swift) or setting
+  `options.overwriteExisting = YES` (in obj-c) when creating the subscription,
+  which will make it update the existing subscription with the same name rather
+  than failing if one already exists with that name.
+* Add an option to include the objects from
+  `RLMLinkingObjects`/`LinkingObjects` properties in sync subscriptions,
+  similarly to how `RLMArray`/`List` automatically pull in the contained
+  objects.
```

Comment:
```suggestion
  objects.
  NOTE: This feature requires Realm Object Server 3.21.0 or later.
```
Add to the above what will happen (how an error can be observed) when used against an older ROS.

Reply:
This should also be mentioned in the Compatibility section.

Reply:
The protocol version bump gives us a hard dependency on ROS 3.21.0 and it'll simply fail to connect at all to older versions, so this particular feature can never fail due to the ROS version being too old.

Reply:
The protocol should be negotiable so unless I'm missing something, the client should be able to connect to any ROS with sync 4.x.

Reply:
That would be 3.20.1, but it wouldn't have the RLMLinkingObjects feature. So it actually seems simpler to require 3.21 and not have to explain the difference between working with ROS 3.20.1 or later, but the LinkingObjects only works with 3.21. Since they (3.20 and 3.21) are so close in time simplicity in communication is likely better.

Reply:
So I suggest we don't mention it here, but just mention it in Compatibility and highlight it at the top as well (I made a suggestion for that.